src/logic/addressBook/store/reducer/index.ts

```diff
@@ -2,13 +2,38 @@ import { Action, handleActions } from 'redux-actions'
 
 import { AddressBookEntry, AddressBookState } from 'src/logic/addressBook/model/addressBook'
 import { ADDRESS_BOOK_ACTIONS } from 'src/logic/addressBook/store/actions'
-import { getEntryIndex } from 'src/logic/addressBook/utils'
+import { getEntryIndex, isValidAddressBookName } from 'src/logic/addressBook/utils'
 import { AppReduxState } from 'src/store'
+import { checksumAddress } from 'src/utils/checksumAddress'
 
 export const ADDRESS_BOOK_REDUCER_ID = 'addressBook'
 
 type Payloads = AddressBookEntry | AddressBookState
 
+const batchLoadEntries = (state, action: Action<AddressBookState>): AddressBookState => {
+  const newState = [...state]
+  const addressBookEntries = action.payload
+  addressBookEntries
+    // exclude those entries with invalid name
+    .filter(({ name }) => isValidAddressBookName(name))
+    .forEach((addressBookEntry) => {
+      const { address, ...rest } = addressBookEntry
+
+      // always checksum the address before storing it
+      const newAddressBookEntry = { address: checksumAddress(address), ...rest }
```

Comment:
Validation and checking the sum should be done in the UI code where errors can be caught and displayed.
You already do this in the Import component.